src/graphics/gameplay/board.c

```diff
@@ -0,0 +1,81 @@
+#include "board.h"
+
+#include <stdlib.h>
+#include <stdbool.h>
+
+#include "../include/raylib.h"
+#include "../include/raymath.h"
+
+#include "../tmp_structs.h"
+#include "../../game_functions/board_functions.h"
+#include "constants.h"
+
+int handleBoardActionEvents (Board *board, Vector2 mouse_position, double width) {
+    float tile_width = ((float) width) / board->cols;
+
+    int selected_tile_row = (int) (mouse_position.y / tile_width);
+    int selected_tile_col = (int) (mouse_position.x / tile_width);
+
+    int selected_tile_index = calcIndexFromPosition(selected_tile_row,  selected_tile_col, board->cols);
+    bool gameEnd = false;
+
+    if (IsMouseButtonPressed(MOUSE_BUTTON_LEFT)) {
+        gameEnd = board->click(board, selected_tile_row, selected_tile_col);
+    } else if (IsMouseButtonPressed(MOUSE_BUTTON_RIGHT)) {
+        Tile *tile = board->tile_array + selected_tile_index;
+        tile->is_flagged = !tile->is_flagged;
+    }
+
+    return EXIT_SUCCESS;
+}
+
+
+int drawBoard (Board *board, AssetManager *assets, double x_offset, double y_offset, double width, double height) {
+
+    float tile_width = ((float) width) / board->cols ;
+    float scale = tile_width / TILE_IMAGE_SIZE;
+    Vector2 position;
+    Tile cur_tile;
+    Texture2D asset_to_draw;
+    
+    for (int i = 0; i < board->rows; ++i) {
+        for (int j =0; j < board->cols; ++j) {
```

Comment:
I think it's inefficient to use a double for loop here, especially because drawBoard will be called a lot during the course of the game. (Although we will most likely utilize a smaller board, it might not be an issue)  Instead, you can utilize a similar tactic you used in the ``initializeTileArray`` where you loop through the length of the tile array (or rows times columns) and divide the index by the number of rows to get ``i`` and mod the index by the rows to get ``j``. I understand that you have utilized the pre-increment of ``++i``  and  ``++j`` instead of post increment but I believe it is not hard to translate this into the single for loop (unless I am missing something). From there the position can be calculated as you have and the function works as is. Unless I am missing something this would be an efficient change

Reply:
Wait, I might be stupid. I think that these would both be O(n^2) anyways....

Reply:
It's going to be the same, and it's O(n), not O(n^2) (where n is the total number of tiles.) You want to think about the number of times each tile is accessed (exactly once in this case.)

Reply:
Oh, I see, but wouldn't O(NxM) also make sense where those represent the number of rows and columns?